### PR TITLE
added multiple rivers to SiteModel  with lists incl test

### PR DESCRIPTION
**Pull request by GundulaW**
closing https://github.com/Deltares/FloodAdapt/issues/157

### CODE REVIEW

**Comment by GundulaW** [comment]
only the test_benefits failed for me, which has probably nothing to do with the multiple rivers but with my database. Can the reviewer please run the test?

File: flood_adapt/integrator/sfincs_adapter.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def get_wl_df_from_offshore_his_results(self) -> pd.DataFrame:
         )
         return wl_df
 
-    def add_dis_bc(self, df_ts: pd.DataFrame):
+    def add_dis_bc(self, list_df: list):
```
Comment (dladrichem) [inline flood_adapt/integrator/sfincs_adapter.py:218]: List[pd.DataFrame]

File: flood_adapt/integrator/sfincs_adapter.py [hunk 2]
```diff
@@ -225,17 +225,18 @@ def add_dis_bc(self, df_ts: pd.DataFrame):
         """
 
         # Determine bnd points from reference overland model
-        gdf_locs = self.sf_model.forcing["dis"].vector.to_gdf()
-        gdf_locs.crs = self.sf_model.crs
+        # ASSUMPTION: Order of the rivers is the same as the site.toml file
+        if list_df is not None:
```
Comment (dladrichem) [inline flood_adapt/integrator/sfincs_adapter.py:229]: if list_df is enough

File: flood_adapt/object_model/hazard/event/historical_hurricane.py [hunk 3]
```diff
@@ -58,9 +58,9 @@ def load_file(filepath: Union[str, os.PathLike]):
 
         # load toml into object
         obj.attrs = HistoricalHurricaneModel.parse_obj(toml)
-        if obj.attrs.river.source == "timeseries":
-            river_csv_path = Path(Path(filepath).parents[0], "river.csv")
-            obj.dis_ts = HistoricalHurricane.read_csv(river_csv_path)
+        # if obj.attrs.river.source == "timeseries":
```
Comment (dladrichem) [inline flood_adapt/object_model/hazard/event/historical_hurricane.py:61]: Dont keep commented code

File: flood_adapt/object_model/hazard/event/historical_nearshore.py [hunk 4]
```diff
@@ -29,17 +29,30 @@ def load_file(filepath: Union[str, os.PathLike]):
...
         if obj.attrs.wind.source == "timeseries":
-            wind_csv_path = Path(Path(filepath).parents[0], "wind.csv")
+            wind_csv_path = Path(
+                Path(filepath).parents[0], obj.attrs.wind.timeseries_file
+            )
             obj.wind_ts = HistoricalNearshore.read_csv(wind_csv_path)
-        if obj.attrs.river.source == "timeseries":
-            river_csv_path = Path(Path(filepath).parents[0], "river.csv")
-            obj.dis_ts = HistoricalNearshore.read_csv(river_csv_path)
+        # obj.dis_dict = {"river": [], "data": []}
```
Comment (dladrichem) [inline flood_adapt/object_model/hazard/event/historical_nearshore.py:44]: Dont keep commented code

File: flood_adapt/object_model/hazard/event/historical_offshore.py [hunk 5]
```diff
@@ -29,9 +29,9 @@ def load_file(filepath: Union[str, os.PathLike]):
         if obj.attrs.wind.source == "timeseries":
             wind_csv_path = Path(Path(filepath).parents[0], "wind.csv")
             obj.wind_ts = HistoricalOffshore.read_csv(wind_csv_path)
-        if obj.attrs.river.source == "timeseries":
-            river_csv_path = Path(Path(filepath).parents[0], "river.csv")
-            obj.dis_ts = HistoricalOffshore.read_csv(river_csv_path)
+        # if obj.attrs.river.source == "timeseries":
```
Comment (dladrichem) [inline flood_adapt/object_model/hazard/event/historical_offshore.py:32]: Same

File: flood_adapt/object_model/hazard/hazard.py [hunk 6]
```diff
@@ -371,18 +364,19 @@ def preprocess_sfincs(
...
-            )
-            self.add_discharge()
-            # convert to metric units
-            gui_units = UnitfulDischarge(
-                value=1.0, units=self.site.attrs.gui.default_discharge_units
-            )
-            conversion_factor = gui_units.convert(UnitTypesDischarge("m3/s"))
-            self.dis_ts = conversion_factor * self.dis_ts
-            model.add_dis_bc(self.dis_ts)
+            # ASSUMPTION: Order of the rivers is the same as the site.toml file
```
Comment (dladrichem) [inline flood_adapt/object_model/hazard/hazard.py:367]: Cant you check this? Check if the lists are the same

File: site.toml
Comment (dladrichem) [inline site.toml:1]: Do we need this file in the backend?

File: tests/test_database/charleston/input/events/ETA/ETA.toml [hunk 7]
```diff
@@ -15,7 +15,7 @@ source = "none"
 [rainfall]
```
Comment (dladrichem) [inline tests/test_database/charleston/input/events/ETA/ETA.toml:15]: We grab the test_database from the P drive right? Then why are we making changes to this (or why is it here at all)

Comment (dladrichem) [reply tests/test_database/charleston/input/events/ETA/ETA.toml:15]: The test_database is last edited 4 months ago so no way it is up-to-date

File: site.toml
Comment (GundulaW) [reply site.toml:1]: yes, for the tests

File: flood_adapt/integrator/sfincs_adapter.py (hunk 2 above)
Comment (GundulaW) [reply flood_adapt/integrator/sfincs_adapter.py:229]: That's not enough but the following works:
if np.any(list_df):

File: flood_adapt/object_model/hazard/hazard.py (hunk 6 above)
Comment (GundulaW) [reply flood_adapt/object_model/hazard/hazard.py:367]: I included a check in the SFINCS adapter that will throw a value error if the river coordinates are not the same or not in the same order. 
I also included that in the test test_hazard_preprocess_synthetic_discharge

**Pull request merged**